syntax = "proto3";

message Product {
  int32 id = 1;
  string name = 2;
  string description = 3;
  float price = 4;
  int32 category_id = 5;
  int32 stock_quantity = 6;
  int32 image_id = 7;
  string created_at = 8;
  string updated_at = 9;
}

message ProductCreate {
  int32 id = 1;
  string name = 2;
  string description = 3;
  float price = 4;
  int32 category_id = 5;
  int32 stock_quantity = 6;
  int32 image_id = 7;
}

message ProductUpdate {
  int32 id = 1;
  string name = 2;
  string description = 3;
  float price = 4;
  int32 category_id = 5;
  int32 stock_quantity = 6;
  int32 image_id = 7;
}

message ProductDelete {
  int32 id = 1;
}

message ProductEvent {
  enum EventType {
    CREATED = 0;
    UPDATED = 1;
    DELETED = 2;
  }

  int32 product_id = 1;
  EventType event_type = 2;
  string event_payload = 3; // JSON-encoded product data
}

message ProductList {
  repeated Product products = 1;
}
